### PR TITLE
Bump the bootstrap compiler to 1.26.0 beta

### PR DESCRIPTION
**Pull request by alexcrichton**
Holy cow that's a lot of `cfg(stage0)` removed and a lot of new stable language
features!

### CODE REVIEW

File: src/libcore/lib.rs [hunk 1]
```diff
@@ -103,9 +101,6 @@
 #![feature(untagged_unions)]
 #![feature(unwind_attributes)]
 
-#![cfg_attr(stage0, allow(unused_attributes))]
-#![cfg_attr(stage0, feature(never_type))]
```
Comment (nikomatsakis) [inline src/libcore/lib.rs]: Given https://github.com/rust-lang/rust/issues/49593, do we still want this?

Comment (alexcrichton) [reply src/libcore/lib.rs]: Perhaps not yeah, but we can always add this back if the never type gets destabilized

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/362411959). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:00:54] configure: rust.quiet-tests     := True
---
[00:02:23] error[E0425]: cannot find function `read_string` in module `fs`
[00:02:23]     --> bootstrap/lib.rs:1152:16
[00:02:23]      |
[00:02:23] 1152 |         t!(fs::read_string(path))
[00:02:23]      |                ^^^^^^^^^^^ did you mean `read_to_string`?
[00:02:23]
[00:02:27] error: aborting due to previous error
[00:02:27]
[00:02:27] For more information about this error, try `rustc --explain E0425`.
[00:02:27] error: Could not compile `bootstrap`.
[00:02:27]
[00:02:27] To learn more, run the command again with --verbose.
[00:02:27] failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml --locked
[00:02:27] Build completed unsuccessfully in 0:01:32
[00:02:27] Makefile:81: recipe for target 'prepare' failed
[00:02:27] make: *** [prepare] Error 1
[00:02:27] Command failed. Attempt 2/5:
[00:02:27]    Compiling bootstrap v0.0.0 (file:///checkout/src/bootstrap)
[00:02:29] error[E0425]: cannot find function `read_string` in module `fs`
[00:02:29]     --> bootstrap/lib.rs:1152:16
[00:02:29]      |
[00:02:29] 1152 |         t!(fs::read_string(path))
[00:02:29]      |                ^^^^^^^^^^^ did you mean `read_to_string`?
[00:02:29]
[00:02:33] error: aborting due to previous error
[00:02:33]
[00:02:33] For more information about this error, try `rustc --explain E0425`.
[00:02:33] error: Could not compile `bootstrap`.
[00:02:33]
[00:02:33] To learn more, run the command again with --verbose.
[00:02:33] failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml --locked
[00:02:33] Build completed unsuccessfully in 0:00:06
[00:02:33] Makefile:81: recipe for target 'prepare' failed
[00:02:33] make: *** [prepare] Error 1
[00:02:33] Command failed. Attempt 3/5:
[00:02:34]    Compiling bootstrap v0.0.0 (file:///checkout/src/bootstrap)
[00:02:35] error[E0425]: cannot find function `read_string` in module `fs`
[00:02:35]     --> bootstrap/lib.rs:1152:16
[00:02:35]      |
[00:02:35] 1152 |         t!(fs::read_string(path))
[00:02:35]      |                ^^^^^^^^^^^ did you mean `read_to_string`?
[00:02:35]
[00:02:39] error: aborting due to previous error
[00:02:39]
[00:02:39] For more information about this error, try `rustc --explain E0425`.
[00:02:39] error: Could not compile `bootstrap`.
[00:02:39]
[00:02:39] To learn more, run the command again with --verbose.
[00:02:39] failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml --locked
[00:02:39] Build completed unsuccessfully in 0:00:05
[00:02:39] Makefile:81: recipe for target 'prepare' failed
[00:02:39] make: *** [prepare] Error 1
[00:02:39] Command failed. Attempt 4/5:
[00:02:39]    Compiling bootstrap v0.0.0 (file:///checkout/src/bootstrap)
[00:02:41] error[E0425]: cannot find function `read_string` in module `fs`
[00:02:41]     --> bootstrap/lib.rs:1152:16
[00:02:41]      |
[00:02:41] 1152 |         t!(fs::read_string(path))
[00:02:41]      |                ^^^^^^^^^^^ did you mean `read_to_string`?
[00:02:41]
[00:02:45] error: aborting due to previous error
[00:02:45]
[00:02:45] For more information about this error, try `rustc --explain E0425`.
[00:02:45] error: Could not compile `bootstrap`.
[00:02:45]
[00:02:45] To learn more, run the command again with --verbose.
[00:02:45] failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --manifest-path /checkout/src/bootstrap/Cargo.toml --locked
[00:02:45] Build completed unsuccessfully in 0:00:05
[00:02:45] make: *** [prepare] Error 1
[00:02:45] Makefile:81: recipe for target 'prepare' failed
[00:02:45] Command failed. Attempt 5/5:
[00:02:45]    Compiling bootstrap v0.0.0 (file:///checkout/src/bootstrap)
[00:02:47] error[E0425]: cannot find function `read_string` in module `fs`
[00:02:47]     --> bootstrap/lib.rs:1152:16
[00:02:47]      |
[00:02:47] 1152 |         t!(fs::read_string(path))
[00:02:47]      |                ^^^^^^^^^^^ did you mean `read_to_string`?
```

</details><p></p>

I'm a bot! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**.

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/362637562). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:00:52] configure: rust.quiet-tests     := True
---
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/boxed.rs:243:13
[00:03:45]     |
[00:03:45] 243 | unsafe impl<#[may_dangle] T: ?Sized> Drop for Box<T> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/arc.rs:903:13
[00:03:45]     |
[00:03:45] 903 | unsafe impl<#[may_dangle] T: ?Sized> Drop for Arc<T> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/rc.rs:808:13
[00:03:45]     |
[00:03:45] 808 | unsafe impl<#[may_dangle] T: ?Sized> Drop for Rc<T> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/raw_vec.rs:712:13
[00:03:45]     |
[00:03:45] 712 | unsafe impl<#[may_dangle] T, A: Alloc> Drop for RawVec<T, A> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/btree/map.rs:141:13
[00:03:45]     |
[00:03:45] 141 | unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for BTreeMap<K, V> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/btree/map.rs:141:30
[00:03:45]     |
[00:03:45] 141 | unsafe impl<#[may_dangle] K, #[may_dangle] V> Drop for BTreeMap<K, V> {
[00:03:45]     |                              ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]    --> liballoc/linked_list.rs:791:13
[00:03:45]     |
[00:03:45] 791 | unsafe impl<#[may_dangle] T> Drop for LinkedList<T> {
[00:03:45]     |             ^^^^^^^^^^^^^
[00:03:45]     |
[00:03:45]     = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]     --> liballoc/vec.rs:2129:13
[00:03:45]      |
[00:03:45] 2129 | unsafe impl<#[may_dangle] T> Drop for Vec<T> {
[00:03:45]      |             ^^^^^^^^^^^^^
[00:03:45]      |
[00:03:45]      = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]     --> liballoc/vec.rs:2427:13
[00:03:45]      |
[00:03:45] 2427 | unsafe impl<#[may_dangle] T> Drop for IntoIter<T> {
[00:03:45]      |             ^^^^^^^^^^^^^
[00:03:45]      |
[00:03:45]      = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error[E0658]: attributes on type parameter bindings are experimental (see issue #34761)
[00:03:45]   --> liballoc/vec_deque.rs:76:13
[00:03:45]    |
[00:03:45] 76 | unsafe impl<#[may_dangle] T> Drop for VecDeque<T> {
[00:03:45]    |             ^^^^^^^^^^^^^
[00:03:45]    |
[00:03:45]    = help: add #![feature(generic_param_attrs)] to the crate attributes to enable
[00:03:45]
[00:03:45] error: aborting due to 10 previous errors
[00:03:45]
[00:03:45] For more information about this error, try `rustc --explain E0658`.
[00:03:45] error: Could not compile `alloc`.
[00:03:45]
[00:03:45] Caused by:
[00:03:45]   process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustc --crate-name alloc liballoc/lib.rs --color always --error-format json --crate-type lib --emit=dep-info,link -C opt-level=2 -C metadata=29528069f2a9d6eb -C extra-filename=-29528069f2a9d6eb --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/release/deps --extern core=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/deps/libcore-1b8789e893adb899.rlib --extern std_unicode=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-std/x86_64-unknown-linux-gnu/release/deps/libstd_unicode-305fbd4ba0177b74.rlib` (exit code: 101)
[00:03:45] warning: build failed, waiting for other jobs to finish...
[00:03:51] error: build failed
[00:03:51] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "--release" "--locked" "--color" "always" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "--message-format" "json"
[00:03:51] expected success, got: exit code: 101
[00:03:51] thread 'main' panicked at 'cargo must succeed', bootstrap/compile.rs:1075:9
[00:03:51] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:03:51] travis_fold:end:stage0-std
[00:03:51] travis_time:end:stage0-std:start=1522937352694087486,finish=1522937388498677220,duration=35804589734
[00:03:51] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/tools/tidy
[00:03:51] Build completed unsuccessfully in 0:00:37
[00:03:51] make: *** [tidy] Error 1
[00:03:51] Makefile:79: recipe for target 'tidy' failed
---
$ ls -lat $HOME/Library/Logs/DiagnosticReports/
ls: cannot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
```

</details><p></p>

I'm a bot! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**.

**Comment by nikomatsakis** [comment]
r=me then

**Pull request merged**